### ISSUE TITLE
is there any way to disable captive portal page which is opening at connect to open wifi?

### ISSUE THREAD

**Issue by vb-star** [alternadom/WiFiFlutter#120]

**Comment by Niek** [comment]
Same issue here. When manually connecting to the WiFi network, it works fine. When using `connect()/findAndConnect()`, all HTTP requests to internal IPs hang because of the captive portal check.

**Comment by daadu** [comment]
This should not happen since we call `.removeCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)` which removes the requirement of the network to have internet.

@Niek  I think we need to enforce the process to use this network, once conneted.

Can you add `ConnectivityManager.setProcessDefaultNetwork(network);` in `onAvailable` callback of `networkCallback` and test if the captive portal still comes?

**Comment by daadu** [comment]
in this case also need to `ConnectivityManager.setProcessDefaultNetwork(null);` in `disconnect`.

**Comment by Niek** [comment]
@daadu that worked! thanks!

Do you want me to make a PR?

**Comment by daadu** [comment]
Yes, please file the PR, make sure `ConnectivityManager.setProcessDefaultNetwork(null);` is called to clean up.

**Comment by daadu** [comment]
`ConnectivityManager.setProcessDefaultNetwork` is [deprecated](https://developer.android.com/reference/android/net/ConnectivityManager#setProcessDefaultNetwork(android.net.Network)) and therefore not reliable.

Use `connManger. bindProcessToNetwork` instead.

**Comment by daadu** [comment]
Infact this is what `forceWifiUsage` does.

I think the "captive portal" wont come if you call that after connect.

**Comment by daadu** [comment]
But I think we should this internally, we have got lot of issues stating the same, after `connect`, the network does not work, I ask them to call `forceWifiUsage` explicitly.

**Comment by Niek** [comment]
Ahhhh.... this works indeed:
```dart
await WiFiForIoTPlugin.forceWifiUsage(true);
```

So there's no need for a PR... maybe this should be documented somewhere though.

**Comment by vb-star** [comment]
can you please check it with android 9 also?

one more thing, 
whether you add  `.removeCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)`  then is your internet is working out of the application in Android 10 onward?

**Comment by Niek** [comment]
Yes, internet outside of the app is working with `withInternet: false`.

But it seems that once I disconnect from the SSID (for example back to the original SSID) I don't have internet anymore in the app. Outside the app it works fine - this is maybe related to #130.

**Comment by vb-star** [comment]
whenever I tried it with my open wifi, yes I have connected to that open wifi but my internet is not working out of the application. 
is there any specific reason?

**Comment by daadu** [comment]
According to the docs the network connected, will only work with the application.

So will not have effect outside the app.

**Comment by daadu** [comment]
> Ahhhh.... this works indeed:
> ```dart
> await WiFiForIoTPlugin.forceWifiUsage(true);
> ```
> 
> So there's no need for a PR... maybe this should be documented somewhere though.

I suggest, this should happen out of the box, instead of documenting it somewhere.

**Comment by vb-star** [comment]
I aware of that, I am also using this but My internet is not working out of the Application. yes, it's working only inner app.

**Comment by daadu** [comment]
> Yes, internet outside of the app is working with `withInternet: false`.
> 
> But it seems that once I disconnect from the SSID (for example back to the original SSID) I don't have internet anymore in the app. Outside the app it works fine - this is maybe related to #130.

This is because with enabling `forceWifiUsage` (internally `bindNetwork`), you need to make sure the bind is removed (properly cleaned up) with disabling it.

This is mandatory (to disable) according to docs, disconnecting will not release the bind.

**Comment by vb-star** [comment]
is there any method for remove binding?

**Comment by daadu** [comment]
> is there any method for remove binding?

`forceWifiUsage(false)`

**Comment by vb-star** [comment]
oh.. sorry, Let me check with this.

Actually, I have done it with some modification in deprecate code and It's working well. 
My internet is working out of application too.
